src/algebra/algebra/subalgebra.lean

```diff
@@ -436,6 +436,18 @@ variables {R : Type u} {A : Type v}
 variables [comm_semiring R] [semiring A] [algebra R A]
 variables (S : subalgebra R A)
 
+instance [subsingleton A] : subsingleton (subalgebra R A) :=
+⟨λ B C, ext (λ x, by { simp only [subsingleton.elim x 0, zero_mem] })⟩
+
+instance [subsingleton (subalgebra R A)] : subsingleton (A ≃ₐ[R] A) :=
+⟨begin
+  intros f g,
+  ext a,
+  rcases set.mem_range.mp (mem_bot.mp (show a ∈ (⊥ : subalgebra R A), from eq_bot_iff.mp
+    (subsingleton.elim (⊤ : subalgebra R A) ⊥) mem_top)) with ⟨x, rfl⟩,
```

Comment:
This looks suspiciously like the proof I use in my other PR to obtain a proof of `subsingleton A`. If so, should that just be the hypothesis?

Reply:
I'm not sure what you mean. Are you saying that the hypothesis of this instance should be `subsingleton A`?

Reply:
I'm saying that might lead to an easier proof, and I think is an equivalent statement

Reply:
I don't think that `subsingleton A` is equivalent to `subsingleton (subalgebra R A)`. For instance, let `R` be the field of rational numbers and let `A` be the field of rational numbers. Then `A` is not `subsingleton`, but `subalgebra R A` is subsingleton.

The fact that `subsingleton A` implies `subsingleton (subalgebra R A)` is the instance above this one.

Reply:
Nicely explained, I retract my comment. I guess this didn't come up for me in the monoid case because there weren't two types to consider.